src/ServiceInsight/MessageFlow/MessageFlowView.xaml

```diff
@@ -364,32 +364,68 @@
+                                                                           Foreground="{StaticResource MessageBackground}"
+                                                                           Padding="3 0"
+                                                                           Text="{Binding SagaType}"
+                                                                           TextTrimming="CharacterEllipsis"
+                                                                           TextWrapping="NoWrap" />
+                                                                    </Border>
+                                                                </Grid>
+                                                                <DataTemplate.Triggers>
+                                                                    <DataTrigger Binding="{Binding IsSagaInitiated}"
+                                                                                 d:DataContext="{d:DesignInstance local:SagaInfo}"
+                                                                                 Value="True">
+                                                                        <Setter TargetName="SagaStatusIcon" Property="Template" Value="{StaticResource SagaTriggerIcon}" />
+                                                                        <Setter TargetName="SagaStatusIcon" Property="ToolTip" Value="Message originated from Saga" />
+                                                                    </DataTrigger>
+                                                                    <MultiDataTrigger d:DataContext="{d:DesignInstance local:SagaInfo}">
+                                                                        <MultiDataTrigger.Conditions>
+                                                                            <Condition Binding="{Binding IsSagaInitiated}" Value="False" />
+                                                                            <Condition Binding="{Binding IsSagaCompleted}" Value="False" />
+                                                                        </MultiDataTrigger.Conditions>
+                                                                        <Setter TargetName="SagaStatusIcon" Property="Template" Value="{StaticResource SagaInitiatedIcon}" />
+                                                                        <Setter TargetName="SagaStatusIcon" Property="ToolTip" Value="Saga Initiated / Updated" />
+                                                                    </MultiDataTrigger>
+                                                                    <MultiDataTrigger d:DataContext="{d:DesignInstance local:SagaInfo}">
+                                                                        <MultiDataTrigger.Conditions>
+                                                                            <Condition Binding="{Binding IsSagaInitiated}" Value="False" />
```

Comment:
Can a saga be initiated and completed in the same message? If so, wouldn't this show the trigger icon?

Reply:
maybe...but didn't change any logic there :)